internal/internal_worker_base.go

```diff
@@ -365,6 +382,23 @@ func (bw *baseWorker) runTaskDispatcher() {
 	}
 }
 
+func (bw *baseWorker) runEagerTaskDispatcher() {
+	defer bw.stopWG.Done()
+	for {
+		select {
+		case <-bw.stopCh:
+			// drain eager dispatch queue
+			for len(bw.eagerTaskQueueCh) > 0 {
+				eagerTask := <-bw.eagerTaskQueueCh
```

Comment:
Poll task dispatch does not do this, it just drops polled tasks on shutdown.

Reply:
I'll note `tryReserveSlot`  will return false if we are closing so this channel will only shrink once we are closing

Reply:
There is still an edge case that if the worker shuts down between `tryReserveSlot` and when `pushEagerTask` it may not get processed

Reply:
I wonder if the eager task receive should just be on the regular `runTaskDispatcher`? Doesn't matter except for keeping the logic close. But no benefit to having this run separately I don't think.

> Poll task dispatch does not do this, it just drops polled tasks on shutdown.

Wonder if we should open another issue to fix this

Reply:
>Wonder if we should open another issue to fix this

Yes I'll open an issue for that

>I wonder if the eager task receive should just be on the regular runTaskDispatcher? Doesn't matter except for keeping the logic close. But no benefit to having this run separately I don't think.

I was trying to keep the behavior the same with this change. While I don't think it's likely putting them together would effect performance, we don't really have the tools to test it right now.